### PR TITLE
Improve accelerometer graph

### PR DESCRIPTION
**Pull request by bolderkat**
Fixed inverted axes, added reset button for max G force labels, and rate limited point updates to 15 Hz to dampen jitter.

### CODE REVIEW

File: Track Buddy/MotionManager.swift [hunk 1]
```diff
@@ -12,32 +12,57 @@ import Collections
 import CoreGraphics
 
 class MotionManager: ObservableObject {
+    private enum Parameters {
+        static let deviceMotionUpdateInterval: TimeInterval = 1/100
+        static let graphUpdateInterval = RunLoop.SchedulerTimeType.Stride(1/15)
+        static var pointStorageLimit: Int {
+            Int(1 / graphUpdateInterval.magnitude * 3)
+        } // final number after multiplier == number of seconds retained for tracer line
```
Comment (lokae0) [inline Track Buddy/MotionManager.swift]: nit- comment should go above line 19

Comment (lokae0) [reply Track Buddy/MotionManager.swift]: I get what you're going for, but it's not very common. 

Could be misconstrued as "should there be another / missing line after this comment?"

File: Track Buddy/MotionManager.swift [hunk 2]
```diff
@@ -12,32 +12,57 @@ import Collections
...
     
     /* With device oriented vertically, axes are:
-     +Z - braking
-     -Z - acceleration
-     +X - right
-     -X - left
+     +Z - acceleration
+     -Z - braking
+     +X - left
+     -X - right
```
Comment (lokae0) [inline Track Buddy/MotionManager.swift:42]: Good commenting!!

File: Track Buddy/MotionManager.swift [hunk 3]
```diff
@@ -12,32 +12,57 @@ import Collections
...
+    
     
     // Max G force values
     @Published private(set) var maxBraking: Double = 0.0
     @Published private(set) var maxAcceleration: Double = 0.0
     @Published private(set) var maxRight: Double = 0.0
     @Published private(set) var maxLeft: Double = 0.0
+
+    
+    // Rate-limited point that updates at graphUpdateInterval to smooth out graph movement
```
Comment (lokae0) [inline Track Buddy/MotionManager.swift]: Just a suggestion, you can use backticks to denote code. `graphUpdateInterval`

Screen Shot 2021-07-01 at 12 41 08

For /// doc comments, these show up as "code" formatted also.

File: Track Buddy/MotionManager.swift [hunk 4]
```diff
@@ -12,32 +12,57 @@ import Collections
...
 class MotionManager: ObservableObject {
+    private enum Parameters {
+        static let deviceMotionUpdateInterval: TimeInterval = 1/100
+        static let graphUpdateInterval = RunLoop.SchedulerTimeType.Stride(1/15)
+        static var pointStorageLimit: Int {
+            Int(1 / graphUpdateInterval.magnitude * 3)
+        } // final number after multiplier == number of seconds retained for tracer line
+    }
+    
+    // Expose graph update interval to Views that need it
```
Comment (lokae0) [inline Track Buddy/MotionManager.swift]: This comment is not very useful. Should be more descriptive of what it actually is, not where it's used externally (as that can always change).

For comments above public / internal properties, get used to doing doc comments (///). These show up when you option-click a property or function. Write as if you're producing a public API / framework. 

(Definitely earns points in interviews btw, whether take-home or pairing)

File: Track Buddy/MotionManager.swift [hunk 5]
```diff
@@ -74,32 +91,52 @@ class MotionManager: ObservableObject {
...
     
-    // TODO: a good place to use @MainActor here
+    private func setUpThrottledPointPublisher() {
+        let cancellable = pointPublisher
+            .throttle(for: Parameters.graphUpdateInterval, scheduler: RunLoop.main, latest: true)
+            .receive(on: DispatchQueue.main, options: nil)
+            .sink(receiveValue: { [weak self] point in
+                self?.throttledPoint = point
+                self?.addToDeque(with: point)
+            })
```
Comment (lokae0) [inline Track Buddy/MotionManager.swift]: Totally fine as-is, but sinks are a pretty good candidate for trailing closures as the usage is common enough for clarity.

```
.sink { ...
}
```

File: Track Buddy/New Group/AccelerometerView.swift [hunk 6]
```diff
@@ -67,6 +68,17 @@ struct AccelerometerView: View {
...
     }
     
+    private var resetButton: some View {
+        Button {
+            motionManager.resetMaxValues()
+        } label: {
+            Text("Reset")
+        }
+        .padding()
+        .background(RoundedRectangle(cornerRadius: 20.0))
```
Comment (lokae0) [inline Track Buddy/New Group/AccelerometerView.swift]: Magic number should be in a Metrics enum in this view 
(don't need that "drawing constants" comment for all of these btw)

File: Track Buddy/MotionManager.swift [hunk 7]
```diff
@@ -12,32 +12,57 @@ import Collections
 import CoreGraphics
 
 class MotionManager: ObservableObject {
+    private enum Parameters {
```
Comment (lokae0) [inline Track Buddy/MotionManager.swift:15]: Nice

**Comment by lokae0** [comment]
Guess it's a good sign most of my comments are just about comments and minor style things :p

**Comment by lokae0** [comment]
Also, gifs in the PR description are great (you had one in the last PR)

**Comment by bolderkat** [comment]
Thanks for the review and feedback! I had thought about doing a gif for this one but decided against it because it wouldn’t look much different from the last one cuz they generally have low framerates…

File: Track Buddy/MotionManager.swift (hunk 2 above)
Comment (bolderkat) [reply Track Buddy/MotionManager.swift:42]: :)

**Pull request merged**